### PR TITLE
[Merged by Bors] - feat(Mathlib/Topology/Instances/NNReal): add `IsOrderBornology ℝ≥0`

### PR DESCRIPTION
**Pull request by yoh-tanimoto**
Add instance `IsOrderBornology ℝ≥0`.

motivation: I will need in #20040 (or a later PR) that the range of a compactly supported `NNReal`-valued function is bounded below and above.

see also https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/IsOrderBornology.20.E2.84.9D.E2.89.A50

### CODE REVIEW

File: Mathlib/Topology/Instances/NNReal.lean [hunk 1]
```diff
@@ -23,12 +23,15 @@ Instances for the following typeclasses are defined:
...
 * `OrderTopology ℝ≥0`
+* `IsOrderBornology ℝ≥0`
 * `ProperSpace ℝ≥0`
 * `ContinuousSub ℝ≥0`
 * `HasContinuousInv₀ ℝ≥0` (continuity of `x⁻¹` away from `0`)
 * `ContinuousSMul ℝ≥0 α` (whenever `α` has a continuous `MulAction ℝ α`)
 
 Everything is inherited from the corresponding structures on the reals.
+(TODO: As for `IsOrderBornology`, there might be a general argument to prove it for a subtype with
+appropriate assumtions)
```
Comment (eric-wieser) [inline Mathlib/Topology/Instances/NNReal.lean:34]: Let's put this by the `instance`

Comment (yoh-tanimoto) [reply Mathlib/Topology/Instances/NNReal.lean:34]: could you explain what you mean? Should I write "As for `instance : IsOrderBornology ℝ≥0`,... "?

File: Mathlib/Topology/Instances/NNReal.lean [hunk 2]
```diff
@@ -73,6 +76,17 @@ instance : CompleteSpace ℝ≥0 :=
 
 instance : ContinuousStar ℝ≥0 where
   continuous_star := continuous_id
+
+instance : IsOrderBornology ℝ≥0 where
```
Comment (eric-wieser) [inline Mathlib/Topology/Instances/NNReal.lean:80]: I mean:
```suggestion
-- TODO: generalize this to a broader class of subtypes
instance : IsOrderBornology ℝ≥0 where
```

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**